[13.0]  report_py3o, report_py3o_fusion_server: Migration to 13.0

lmignon opened the pull request:
supersedes #339

lmignon committed:
Move to report_py3o after import from HG

lmignon committed:
Add setup

lmignon committed:
[FIX] View inheritance may not use attribute 'string' as a selector.

lmignon committed:
[IMP] New API + Short Hearder

lmignon committed:
Remove unicode string

lmignon committed:
[IMP] Add the possiblity to get the template from an absolute path on the server

lmignon committed:
[IMP] Allow user to use local fusion

lmignon committed:
Add a simple way to extend the parser context

lmignon committed:
Remove ACL on removed model

lmignon committed:
Improve filetype selection name

lmignon committed:
Fix native flag

faide committed:
fixed missing comma in model definition

lmignon committed:
Add simple tests and fix call to fusion server

lmignon committed:
Fix skip fusion on server when the fusion is local

lmignon committed:
Improve tests

lmignon committed:
Replace openerp element by odoo

alexis-via committed:
Small usability improvements

Replace README.md by README.rst (not finished)

alexis-via committed:
Take into accounts most remarks of @lasley

Remove <data> in views
Protect import of py3o libs
Remove dep on base module
Other small changes

lmignon committed:
Replace plain SQL by orm

lmignon committed:
rename module ir_report to ir_actions_report_xml

lmignon committed:
Add more tests and fixes issues found by tests

lmignon committed:
[FIX] Check constrains only if report_type == 'py3o'

alexis-via committed:
report_py3o: full installation procedure in README

alexis-via committed:
Port report_py3o to Odoo v10

alexis-via committed:
Update version numbers

Rename __openerp__.py to __manifest__.py

alexis-via committed:
PEP8 fix

alexis-via committed:
Update README

lmignon committed:
[IMP] Escape 'False' when rendering value

lmignon committed:
[IMP] Replace old style parser by TransientModel
The goal is to improve the modularity by making the parser a true inheritable odoo model and share part of the code with the 'report' model

Conflicts:
	report_py3o/models/ir_actions_report_xml.py
	report_py3o/models/py3o_report.py
	report_py3o/tests/test_report_py3o.py

alexis-via committed:
FIX double dot in filename

alexis-via committed:
Add access to report_xml object in report

alexis-via committed:
FIX Adapt test suite to my recent commit to avoid double dot on filename extension

Commit:
[FIX] imports

lmignon committed:
[IMP] Minimizes memory consumption

Conflicts:
	report_py3o/models/py3o_report.py

Commit:
* travis.yml
* flake8

alexis-via committed:
Update and improve README instructions to setup py3o

lmignon committed:
[IMP] Allow to override/extend the way we get the fallback template

lmignon committed:
[IMP] report_py3o: Take into account print_report_name
If a Printed Report Name is set on the action report, use it as
downloaded filename
refs #133

lmignon committed:
[FIX] report_py3o: fix exception when report must be saved as attachement
The mehtod  must be called with a list of ids not with a list of browse records

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
[FIX] delete PDF invoice attachment on invoice back to draft (native feature now working with py3o)

lmignon committed:
[IMP] report_py3o: update authors and contributors

lmignon committed:
[IMP] report_py3o: prevent injections when retrieving the template from path

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
report_py3o: bump version following last merge

Commit:
Fix py3o freeze when using server fusion

The template file has been partially read for the parsing of the
expressions, so use the original template data instead.  When the fusion
server receives an incomplete template, we don't receive any answer
back.

Commit:
Pass option for escaping False values to server

Needs https://bitbucket.org/faide/py3o.fusion/pull-requests/1

alexis-via committed:
Add option py3o_multi_in_one for Py3o reports

oca-transbot committed:
OCA Transbot updated translations from Transifex

lmignon committed:
[FIX] report_py3o:  must return the path to the report
refs #119

oca-transbot committed:
OCA Transbot updated translations from Transifex

omar7r committed:
[IMP] report_py3o: Allows to interpret \n and \t in texts, cleans html tags too

omar7r committed:
[FIX] Deletes an unnecessary function call

omar7r committed:
[FIX] report_py3o: Removes api.one

omar7r committed:
[FIX] report_py3o: Import new dependencies in try...except

omar7r committed:
[UPD] report_py3o: Bump version

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
[RFR] split off the fusion server to its own module

use libreoffice for conversions in the base version. Fixes #179

jesusVMayor committed:
[FIX]report_py3o: Escape correctly html characters.

alexis-via committed:
Improve README.rst for report_py3o* modules

hbrunn committed:
[ADD] migration script to install report_py3o_fusion_server on upgrades

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
[FIX] fallback to libreoffice conversion when no server is configured

fixes #200

pedrobaeza committed:
[FIX] report_py3o: Bump version number

hbrunn committed:
[IMP] check for template data, not for the id

alexis-via committed:
report_py3o_fusion_server: Add support for PDF Export options of libreoffice

alexis-via committed:
report_py3o_fusion_server: Update README.rst

Fix string

alexis-via committed:
report_py3o_fusion_server: Add chronometer

Improve logs

alexis-via committed:
Fix wrong default value

SimoRubi committed:
[FIX] If this is not a py3o report, do nothing at all. In some cases the UI remains blocked otherwise

alexis-via committed:
Improve README.rst to help those that want to produce fully valid PDF/A documents

sbidoul committed:
report_py3o: mention formatLang in README

oca-travis committed:
[UPD] Update report_py3o.pot

oca-travis committed:
[UPD] Update report_py3o_fusion_server.pot

kryskool committed:
Translated using Weblate (Français)

Currently translated at 100,0% (38 of 38 strings)

Translation: reporting-engine-10.0/reporting-engine-10.0-report_py3o
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-10-0/reporting-engine-10-0-report_py3o/fr/

lmignon committed:
[MIG] report_py3o, report_py3o_fusion_server: Migration to 12.0

lmignon committed:
[IMP] report_py2o: cleanup js

lmignon committed:
[FIX] report_py3o_fusion_server: requests.post returns bytes not str

Commit:
[12.0][MIG] improvement py3o_report_extender

sbejaoui committed:
Update i18n

lmignon committed:
[FIX] report_py3o: Use the right naming convention for the render method
To determine the method to use to render a recport according to its type, the generic method 'render' defined into ir.action.report check if a method name  is defined on the model. https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/ir_actions_report.py#L734
Therefore, we must provide this method to be compliant with what's expected by Odoo.
This change will also allows the usage of py3o template with mail_template once https://github.com/odoo/odoo/pull/30013 will be merged

lmignon committed:
[IMP] report_py3o: remove obsolete methods

lmignon committed:
[IMP] report_py3o: Document new methods available into the parser context

lmignon committed:
[FIX] report_py3o: Add missing method _merge_pdf.
This method was previously provided by Odoo and is used to merge all the reports generated if the generation is called for more than one record

lmignon committed:
[IMP][FIX] py3o_report, py3o_report_fusion_server: Compute the availability of py3o report
Before this change it was not possible to install modules declaring py3o report into a non native format without specifying a Fusion server once the module py3o_report_fusion_server was installed. With theses changes, we now take care of the availability of the libreoffice runtime to display/log a warning message when the report is in a non native runtime.

lmignon committed:
[FIX] report_py3o: Add missing 'user' and 'lang' into the parser context

lmignon committed:
[IMP] py3o_report: Allow to specify the lang when calling o_format_lang
lang_code is already supported by o_format_date

lmignon committed:
[FIX] report_py3o: Access to ir.config_parameter with sudo

houzefa-abba committed:
[FIX] report_py3o: respect _postprocess_report method signature

Co-Authored-By: lmignon <laurent.mignon@acsone.eu>

oca-travis committed:
[UPD] Update report_py3o.pot

oca-travis committed:
[UPD] Update report_py3o_fusion_server.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-12.0/reporting-engine-12.0-report_py3o
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-12-0/reporting-engine-12-0-report_py3o/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-12.0/reporting-engine-12.0-report_py3o_fusion_server
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-12-0/reporting-engine-12-0-report_py3o_fusion_server/

oca-travis committed:
[UPD] Update report_py3o.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-12.0/reporting-engine-12.0-report_py3o
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-12-0/reporting-engine-12-0-report_py3o/

alexis-via committed:
Update README for report_py3o and report_py3o_fusion_server (#306)

* Update README for report_py3o and report_py3o_fusion_server

Convert README to the new readme subdirectory

sbidoul committed:
[FIX] report_py3o readme structure

OCA-git-bot committed:
[UPD] README.rst

dupski committed:
Make sure value is always set for computed fields
(without these a CacheMiss error is thrown by odoo)

lmignon committed:
[IMP] report_py3o, report_py3o_fusion_server: black, isort

lmignon committed:
[MIG] report_py3o, report_py3o_fusion_server: Migration to 13.0

lmignon committed:
[ADD] setup.py

njeudy commented on report_py3o/README.rst:
```diff
@@ -0,0 +1,244 @@
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Freporting--engine-lightgray.png?logo=github
+    :target: https://github.com/OCA/reporting-engine/tree/12.0/report_py3o
```
```suggestion
    :target: https://github.com/OCA/reporting-engine/tree/13.0/report_py3o
```

njeudy commented on report_py3o/README.rst:
```diff
@@ -0,0 +1,244 @@
+    :target: https://github.com/OCA/reporting-engine/tree/12.0/report_py3o
+    :alt: OCA/reporting-engine
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/reporting-engine-12-0/reporting-engine-12-0-report_py3o
```
```suggestion
    :target: https://translation.odoo-community.org/projects/reporting-engine-13-0/reporting-engine-13-0-report_py3o
```

njeudy commented on report_py3o/README.rst:
```diff
@@ -0,0 +1,244 @@
+    :target: https://translation.odoo-community.org/projects/reporting-engine-12-0/reporting-engine-12-0-report_py3o
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/143/12.0
```
```suggestion
    :target: https://runbot.odoo-community.org/runbot/143/13.0
```

njeudy reviewed (changes requested):
- Ok for me code and test

Just some readme typo.

lmignon commented:
@njeudy The readme file file is generated by oca-gen-addon-readme and will be regenerated by the bot when merging....

sbejaoui committed:
[12.0][FIX] - report_py3o: run libreoffice in an isolated user installation

Bug when more than one conversion print is launched within the same libreoffice instance.

The standard behavior of libreoffice when a user open it while another instance is
running is to show a new window and throw an error if a new instance is forced within
the same user installation [see](https://bugs.documentfoundation.org/show_bug.cgi?id=37531).

This implies a bug in report_py3o module when we call libreoffice at the same time for
different documents.

To reproduce this bug:

**Case 1:**
1.  Simultaneously print two documents.

**Case 2:**
1.  Run print jobs using job_queue module
2.  Manually print another document

**Case 3:**
2.  Open libreoffice
3.  Print a py3o report

This PR creates a temporary user installation for each libreoffice conversion to bypass this limitation.

sudo-san commented:
Is there an approximate date when report_py3o will be available for the V13?

Thank you for a short feedback.

lmignon commented:
@sudo-san These addons are already migrated by reviews are required to get these merged

sudo-san commented:
@lmignon Thanks a lot for you quick reply. Is there already a way to download the current version for odoo 13?

lmignon commented:
> @lmignon Thanks a lot for you quick reply. Is there already a way to download the current version for odoo 13?

from the submitted branch. https://github.com/acsone/reporting-engine/tree/13.0-mig-report_py3o

njeudy commented:
@lmignon any way to make it merged ? I use this branch on my V13 deployments ..

lmignon committed:
Merge branch '13.0' into 13.0-mig-report_py3o

lmignon committed:
[FIX] report_py3o: prettier xml after merge with master

lmignon commented:
@njeudy Before the end of my wok and now some pre-commit hooks were added. After some tests, I'm not able to find a way to fix the code and have pre-commit happy....

lmignon committed:
[FIX] eslint error

lmignon committed:
[FIX] pre-commit: remove obsole 'known_third_party' from isort.cfg

sbidoul committed:
[FIX] make prettier happy

tonygalmiche reviewed (approved):
I tested on my computer
Everything works as expected
Thank you for this migration

lmignon commented:
All the tests works on my computer and I don't see what's wrong with travis.... We also use successfully these addons into production.

dreispt commented:
@lmignon TravicCI log shows one test failing. I can try a merge and see it we get a different result.

dreispt commented:
/ocabot merge nobump

lmignon commented:
Thak you @dreispt
